containers/unit_tests/TestPaddingPropagation.hpp

```diff
@@ -0,0 +1,153 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_DualView.hpp>
+#include <Kokkos_Core.hpp>
+
+#define VIEW_T Kokkos::DualView
+
+#define RANK_0 0
+#define RANK_1 1
+#define RANK_2 2
+#define RANK_3 3
+#define RANK_4 4
+#define RANK_5 5
+#define RANK_6 6
+#define RANK_7 7
+
+#define PARAM_0
+#define PARAM_1 50
+#define PARAM_2 50, 1
+#define PARAM_3 50, 1, 50
+#define PARAM_4 50, 1, 1, 50
+#define PARAM_5 50, 1, 1, 1, 50
+#define PARAM_6 50, 1, 1, 1, 1, 50
+#define PARAM_7 50, 1, 1, 1, 1, 1, 50
+
+namespace Test {
+
+using DType   = int;
+using DType_0 = DType;
+using DType_1 = DType *;
+using DType_2 = DType **;
+using DType_3 = DType ***;
+using DType_4 = DType ****;
+using DType_5 = DType *****;
+using DType_6 = DType ******;
+using DType_7 = DType *******;
+
+#define ASSERT_VIEW_STRIDES(view1, view2, rank)           \
+  (view1.stride(0) == view2.stride(0) || rank < 1) &&     \
```

Comment:
For clarity of control flow, I'd rather each line's condition were written in the opposite order, 
```suggestion
  (rank < 1 || view1.stride(0) == view2.stride(0)) &&     \
```
But I don't see that as a gating condition.